### PR TITLE
STY: Simplify _extract_text

### PR DESCRIPTION
**Pull request by MartinThoma**

### CODE REVIEW

**Comment by MartinThoma** [comment]
@pubpub-zz `_extract_text` is too complex. By moving some code blocks into own functions I hope I can reduce the complexity. The goal is to make it easier to understand how the text extraction works and make it easier to extend / adjust it.

I still want to group some of the parameters, but I'm uncertain at the moment what makes sense. What do you think about this approach?

**Comment by MartinThoma** [comment]
The block in `if check_crlf_space:` could also be a function

**Comment by pubpub-zz** [comment]
As the objective is to have as many people as possible to maintain the code, this is OK for me 😀

File: pypdf/_page.py [hunk 1]
```diff
@@ -2392,3 +2266,192 @@ def _get_fonts_walk(
         _get_fonts_walk(cast(DictionaryObject, obj[key]), fnt, emb)
 
     return fnt, emb  # return the sets for each page
+
+
+def handle_tj(
```
Comment (pubpub-zz) [inline pypdf/_page.py:2271]: Wouldn't it be preferable to have this function local to the page class or at least as an classmethod?
the code will be next to this calling function

File: pypdf/_page.py [hunk 2]
```diff
@@ -2392,3 +2266,192 @@ def _get_fonts_walk(
...
+                        output += text
+                        if visitor_text is not None:
+                            visitor_text(text, cm_matrix, tm_matrix, cmap[3], font_size)
+                        text = ""
+                    text = text + x
+                # fmt: on
+    return text, rtl_dir
+
+
+def crlf_space_check(
```
Comment (pubpub-zz) [inline pypdf/_page.py:2355]: Wouldn't it be preferable to have this function local to the page class or at least as an classmethod?
the code will be next to this calling function

File: pypdf/_page.py (hunk 1 above)
Comment (MartinThoma) [reply pypdf/_page.py]: Why would you prefer that?

I just moved those parts to a separate `_text_extraction` submodule within this PR. I hope that I can separate the text extraction code from the other responsibilities of the page class at some point (not necessarily in this PR)

File: pypdf/_page.py (hunk 2 above)
Comment (MartinThoma) [reply pypdf/_page.py]: (let's keep the discussion in the comment above :-) I'll not forget that it applies to this part as well ... I hope :sweat_smile: :crossed_fingers: )

**Pull request merged**